www/caddy/src/opnsense/mvc/app/models/OPNsense/Caddy/Caddy.php

```diff
@@ -107,6 +108,52 @@ private function checkSubdomainsAgainstDomains($subdomains, $domains, $messages)
         }
     }
 
+    // Get the OPNsense WebGUI configuration and store it
+    private $webgui;
+
+    private function getWebGui() {
+        if (!$this->webgui) {
+            $this->webgui = Config::getInstance()->object()->system->webgui ?? null;
+        }
+        return $this->webgui;
+    }
+
+    // Get the default OPNsense WebGUI ports
+    private function getWebGuiPorts() {
+        $webgui = $this->getWebGui();
+        $webGuiPorts = [];
+
+        // Only add ports to array if no specific interfaces for the WebGUI are set
+        if (empty($webgui->interfaces)) {
+            // Add port 443 if no specific port is set, only checks for default WebGUI settings
+            if (empty($webgui->port)) {
+                $webGuiPorts[] = '443';
+            }
+
+            // Add port 80 if HTTP redirect is not explicitly disabled
+            if (!isset($webgui->disablehttpredirect) || $webgui->disablehttpredirect != '1') {
+                $webGuiPorts[] = '80';
+            }
+        }
+
+        return $webGuiPorts;
+    }
+
+    // 4. Check for conflicts between Caddy and OPNsense WebGUI ports
+    private function checkWebGuiSettings($messages) {
+        $webGuiPorts = $this->getWebGuiPorts();
+        $tlsAutoHttpsSetting = $this->general->TlsAutoHttps->__toString();
+
+        // Since only one generic message is added, the array is checked for not empty. Only add message if AutoHttps is other than off.
+        if (!empty($webGuiPorts) && $tlsAutoHttpsSetting !== 'off') {
+            $messages->appendMessage(new Message(
+                gettext('To use Auto HTTPS, resolve these conflicts: The default ports are currently configured for the OPNsense WebGUI. Go to "System - Settings - Administration" and change "TCP port" to a non-standard port, e.g., 8443. Additionally, enable "Disable web GUI redirect rule".'),
```

Comment:
I would extend the message with the overlapping parts (`implode(',', $overlap)`) for clarity.